File: go/pkg/models/social_account.go
package models

import (
	"time"

	"github.com/google/uuid"
)

type SocialAccount struct {
	ID           uuid.UUID  `gorm:"type:uuid;primary_key;" json:"id"`
	CreatedAt    time.Time  `json:"createdAt"`
	UpdatedAt    time.Time  `json:"updatedAt"`
	UserID       uuid.UUID  `gorm:"type:uuid;index;not null" json:"userId"`
	User         User       `json:"user" gorm:"foreignKey:UserID"`
	Provider     string     `gorm:"type:varchar(32);not null" json:"provider"`
	AccountID    *string    `gorm:"type:varchar(128)" json:"accountId,omitempty"`
	Username     *string    `gorm:"type:varchar(128)" json:"username,omitempty"`
	AccessToken  string     `gorm:"type:text;not null" json:"-"`
	RefreshToken string     `gorm:"type:text" json:"-"`
	ExpiresAt    *time.Time `json:"expiresAt,omitempty"`
	Scopes       *string    `gorm:"type:text" json:"scopes,omitempty"`
}

File: dart/skunkworks/lib/models/social_account.dart
class SocialAccount {
  final String provider;
  final String? accountId;
  final String? username;
  final DateTime? expiresAt;

  SocialAccount({
    required this.provider,
    this.accountId,
    this.username,
    this.expiresAt,
  });

  factory SocialAccount.fromJson(Map<String, dynamic> json) {
    return SocialAccount(
      provider: json['provider'] as String,
      accountId: json['accountId'] as String?,
      username: json['username'] as String?,
      expiresAt: json['expiresAt'] != null
          ? DateTime.tryParse(json['expiresAt'] as String)
          : null,
    );
  }
}
